app/src/main/java/com/example/eric/pokedex/FilterActivity.java

```diff
@@ -0,0 +1,78 @@
+package com.example.eric.pokedex;
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+
+import com.bumptech.glide.Glide;
+
+public class FilterActivity extends AppCompatActivity {
+
+    EditText minHp;
+    EditText minAttack;
+    EditText minDefense;
+
+    Button filterButton;
+    Button getRandom;
+
+    ImageView randomPoke;
+
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_filter);
+
+        minHp = (EditText) findViewById(R.id.minHP);
+        minAttack = (EditText) findViewById(R.id.minAttack);
+        minDefense = (EditText) findViewById(R.id.minDefense);
+
+        filterButton = (Button) findViewById(R.id.submit_filter);
+        getRandom = (Button) findViewById(R.id.getRandomPokemon);
+
+        randomPoke = (ImageView) findViewById(R.id.randomPokeImage);
+        Intent intent = getIntent();
+        String rand_poke_name = intent.getStringExtra("rand_poke");
+
+        String url = "http://img.pokemondb.net/artwork/" + rand_poke_name.toLowerCase() + ".jpg";
+
+        if (rand_poke_name.equalsIgnoreCase("Flabébé")){
```

Comment:
cleaner to use a line to remove all special characters and make lowercase, should be easy to find through stack overflow